src/software/simulated_tests/tbots_test_fixture.py

```diff
@@ -0,0 +1,174 @@
+import time
+
+import decorator
+
+import pytest
+from software.simulated_tests.pytest_main import load_command_line_arguments
+from software.simulated_tests.simulated_test_fixture import simulated_test_initializer
+from software.field_tests.field_test_fixture import field_test_initializer
+import threading
+from software.thunderscope.proto_unix_io import ProtoUnixIO
+from software.thunderscope.thunderscope import Thunderscope
+from software.logger.logger import createLogger
+
+simulator_proto_unix_io = None
+yellow_full_system_proto_unix_io = None
+blue_full_system_proto_unix_io = None
+
+logger = createLogger(__name__)
+
+LAUNCH_DELAY_S = 0.1
+PAUSE_AFTER_FAIL_DELAY_S = 3
+TEST_END_DELAY = 0.5
```

Comment:
Curious what the delays are for? It it to allow thunderscope to launch?

Reply:
`LAUNCH_DELAY_S` is for thunderscope gui to fully launch before the test starts running, `PAUSE_AFTER_FAIL_DELAY_S` is for when the test fails. It gives the user time to look at the final state of the world and validation visualization protos before the gui closes. Similarly, `TEST_END_DELAY_S` gives the user time to see the end worldstate of the simulation